### PR TITLE
chore: type parser

### PR DESCRIPTION
**Pull request by gclapperton**
#### What does this PR do?

Extend typing to cover the parser package. This PR should not change any logic

#### What are the relevant tickets?

[NARW-2131](https://snyksec.atlassian.net/browse/NARW-2131)

[NARW-2131]: https://snyksec.atlassian.net/browse/NARW-2131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/parser/v1.ts [hunk 1]
```diff
@@ -1,8 +1,29 @@
+import {
+  PathObj,
+  Policy,
+  PolicyError,
+  Rule,
+  RuleSet,
+  Severity,
+} from '../types';
+
```
Comment (cat2608) [inline lib/parser/v1.ts:9]: Is it possible that `PolicyError` does not exist in `../types`?

Comment (gclapperton) [reply lib/parser/v1.ts:9]: Oh, yes, it wasn't committed... why doesn't the TS transpiler catch this??? :|

File: lib/parser/v1.ts [hunk 2]
```diff
@@ -11,105 +32,139 @@ function imports(policy) {
...
+      'POLICY_BAD_THRESHOLD'
+    );
   }
 
-  return threshold;
+  return strThreshold;
+}
+
+const isObject = (v: unknown): v is Record<string, unknown> =>
+  Object.prototype.toString.call(v) === '[object Object]'; // typeof returns true for arrays and other types
```
[the remark this answers is not in the document]
Comment (gclapperton) [reply lib/parser/v1.ts:165]: I've pulled this out and added a test in #114

**Review by cat2608 — APPROVED** [review]

**Pull request merged**